adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java

```diff
@@ -38,9 +39,14 @@
 
     private final Function<JsonEnvelope, String> responseBodyGenerator;
 
-    RestProcessor(RestEnvelopeBuilderFactory envelopeBuilderFactory, Function<JsonEnvelope, String> responseBodyGenerator) {
+    private final boolean sendMetadataIdInHeader;
+
+    RestProcessor(final RestEnvelopeBuilderFactory envelopeBuilderFactory,
+                  final Function<JsonEnvelope, String> responseBodyGenerator,
+                  final boolean sendMetadataIdInHeader) {
```

Comment:
Maybe we should pass in a function to add headers, rather than a boolean - to avoid a proliferation of flag arguments